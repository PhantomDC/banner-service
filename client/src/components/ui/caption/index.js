import React, { Component } from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'


const Caption = styled.h1`
  font-family: 'Exo 2', sans-serif !important;
  padding: 0;
  font-size: ${props => props.size};
`
Caption.propTypes = {
  size: PropTypes.number
}

Caption.defaultProps = {
  size: 20,
}

export default Caption
