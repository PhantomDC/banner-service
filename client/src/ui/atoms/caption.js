import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'


export const Caption = styled.h1`
  font-family: 'Myriad' !important;
  padding: 0;
  font-size: ${props => props.size}px;
  color: ${props => props.color};
`

Caption.propTypes = {
  size: PropTypes.number,
  color: PropTypes.string,
}

Caption.defaultProps = {
  size: 20,
  color: 'black',
}
