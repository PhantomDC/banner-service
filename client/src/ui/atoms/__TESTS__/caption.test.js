import React from 'react'
import { configure, render, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { create } from 'react-test-renderer'
import { Caption } from '../caption'

/* eslint-disable no-undef */
/* eslint-disable function-paren-newline */

configure({ adapter: new Adapter() })

describe('test component Caption', () => {
  it('Caption render without crash', () => {
    render(<Caption size={10} />)
  })

  it('size props test', () => {
    const wrapper = shallow(<Caption size={10} />)
    const { size } = wrapper.instance().props

    expect(size).toBe(10)
  })

  it('shapshot Button', () => {
    const tree = create(<Caption size={10} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
