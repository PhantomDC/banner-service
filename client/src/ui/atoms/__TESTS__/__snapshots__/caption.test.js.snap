// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component Caption shapshot Button 1`] = `
<h1
  className="sc-bdVaJa jNWGbQ"
  size={10}
/>
`;
